src/main/scala/org/graphframes/lib/ShortestPaths.scala

```diff
@@ -54,7 +54,7 @@ class ShortestPaths private[graphframes] (private val graph: GraphFrame) extends
    * The list of landmark vertex ids. Shortest paths will be computed to each landmark.
    */
   def landmarks(value: util.ArrayList[Any]): this.type = {
-    landmarks(value.asScala)
+    landmarks(value.asScala.toSeq)
```

Comment:
Why change this ?

Reply:
Looking it a bit closer to why this was needed in probably this change in scala 2.13 (https://github.com/scala/scala/releases/tag/v2.13.0)
```

* Immutable scala.Seq
  * Seq is now an alias for collection.immutable.Seq
    * Before, it was an alias for the possibly-mutable collection.Seq.
```

So when the `java.util.ArrayList[Any]` is converted we end up with a `scala.collection.mutable.Buffer[Any]`. In scala 2.12 this is a sub type of `Seq[Any]` while in scala 2.13 it it not (it only a subtype of mutable.Seq). So my fix of doing `toSeq` probably makes a extra copy. One other solution might be that one change `Seq` to `collection.Seq` in the needed places to keep the 2.12 semantics. With this change
```
diff --git a/src/main/scala/org/graphframes/lib/ShortestPaths.scala b/src/main/scala/org/graphframes/lib/ShortestPaths.scala
index f110d6f..4641ce7 100644
--- a/src/main/scala/org/graphframes/lib/ShortestPaths.scala
+++ b/src/main/scala/org/graphframes/lib/ShortestPaths.scala
@@ -19,6 +19,7 @@ package org.graphframes.lib
 
 import java.util
 
+import scala.collection
 import scala.collection.JavaConverters._
 
 import org.apache.spark.graphx.{lib => graphxlib}
@@ -39,12 +40,12 @@ import org.graphframes.GraphFrame
  *    the shortest-path distance to each reachable landmark vertex.
  */
 class ShortestPaths private[graphframes] (private val graph: GraphFrame) extends Arguments {
-  private var lmarks: Option[Seq[Any]] = None
+  private var lmarks: Option[collection.Seq[Any]] = None
 
   /**
    * The list of landmark vertex ids. Shortest paths will be computed to each landmark.
    */
-  def landmarks(value: Seq[Any]): this.type = {
+  def landmarks(value: collection.Seq[Any]): this.type = {
     // TODO(tjh) do some initial checks here, without running queries.
     lmarks = Some(value)
     this
@@ -54,7 +55,7 @@ class ShortestPaths private[graphframes] (private val graph: GraphFrame) extends
    * The list of landmark vertex ids. Shortest paths will be computed to each landmark.
    */
   def landmarks(value: util.ArrayList[Any]): this.type = {
-    landmarks(value.asScala.toSeq)
+    landmarks(value.asScala)
   }
 
   def run(): DataFrame = {
@@ -64,7 +65,7 @@ class ShortestPaths private[graphframes] (private val graph: GraphFrame) extends
 
 private object ShortestPaths {
 
-  private def run(graph: GraphFrame, landmarks: Seq[Any]): DataFrame = {
+  private def run(graph: GraphFrame, landmarks: collection.Seq[Any]): DataFrame = {
     val idType = graph.vertices.schema(GraphFrame.ID).dataType
     val longIdToLandmark = landmarks.map(l => GraphXConversions.integralId(graph, l) -> l).toMap
     val gx = graphxlib.ShortestPaths.run(
```
it also compile both in 2.12 and 2.13.